src/states/common/approved.rs

```diff
@@ -505,8 +505,13 @@ pub trait Approved: Base {
             info!("{} - ignore Relocate: {:?} - not a member", self, details);
         } else {
             info!("{} - handle Relocate: {:?}.", self, details);
-            self.chain_mut().remove_member(&details.pub_id);
-            self.handle_member_relocated(details, signature, outbox)?;
+
+            // Note: it is important that `handle_member_relocated` is called before the member is
+            // actually removed. This is because the member info of the relocated node might still
+            // be needed by `handle_member_relocated`.
+            let pub_id = details.pub_id;
```

Comment:
I am not really sure I understand, is it because of `get_member_connection_info`?
Ohh it is because of the `node_knowledge` from the Relocating state used by chain::prove.

That implicit dependency seem less than ideal.
Should we have a new Option to pass to the chain::prove function, and get it out of self.chain_mut().remove_member(&pub_id); and pass it down to handle_member_relocated?

Or get the proof in that function and pass it down to handle_member_relocated (still think prove should use optional instead of using that state, that just seem to confusing)?